### PR TITLE
Convert AnswerServer to StatefulSet

### PR DESCRIPTION
**Pull request by lewis-mackay-mf**
Signed-off-by: lewis-mackay-mf

### CODE REVIEW

File: helm/idol-answerserver/Chart.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 # END COPYRIGHT NOTICE
 apiVersion: v2
 name: idol-answerserver
-version: "0.4.0"
+version: "0.5.0"
```
Comment (daniel-lau-mf) [inline helm/idol-answerserver/Chart.yaml:13]:
```suggestion
version: "0.40"
```
Don't need to bump version again yet as we haven't published/release the 0.4.0 version

File: helm/idol-answerserver/templates/_container.tpl [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+{{- define "idolanswerserver.container" -}}
+{{- $root := get . "root" | required "idolanswerserver.container: missing root" -}}
+{{- $component := get . "component" | required "idolanswerserver.container: missing component" -}}
+{{- include "idol-library.util.merge" (dict
+  "root" $root
+  "component" $component
+  "source" "idol-library.aciserver.container.base.v1"
+  "destination" "idolanswerserver.container.base" 
+  "env" (list (dict "name" "IDOL_COMMAND_PARAMS" "value" "-idolcluster"))
+  "volumeMounts" (list (dict "name" "licensing" "mountPath" "/answerserver/license"))
```
Comment (daniel-lau-mf) [inline helm/idol-answerserver/templates/_container.tpl:28]: Will need to persist both the license and uid folders.
Wonder if it's worth making the volume available to use for caching model files too?
Should be possible to use one volume claim and then mount multiple subPaths from within that (see how it's done in the `idol-nifi` chart) - probably want to rename the volumeClaimTemplates to `state` or similar

File: helm/idol-answerserver/values.yaml [hunk 3]
```diff
@@ -31,6 +31,17 @@ idolImage:
   # -- used to construct container image name: {idolImage.registry}/{idolImage.repo}:{idolImage.version}
   repo: "answerserver"
 
+# answerServerVolumeSize -- Size of the PersistentVolumeClaim that is created
+# for each AnswerServer instance. The Kubernetes cluster will need to provide
+# enough PersistentVolumes to satisify the claims made for the desired number
+# of AnswerServer instances.
+answerServerVolumeSize: 16Gi
```
Comment (daniel-lau-mf) [inline helm/idol-answerserver/values.yaml:38]:
```suggestion
answerServerVolumeSize: 2Gi
```
`16Gi` seems excessive for answerserver. Even `2Gi` is a lot for licensing, but if we allow model cache data to go here then 2Gi is probably reasonable.

**Review by daniel-lau-mf — CHANGES_REQUESTED** [review]
See comments

File: helm/release/idol-omnigroupserver-0.6.0.tgz
Comment (daniel-lau-mf) [inline helm/release/idol-omnigroupserver-0.6.0.tgz:1]: This shouldn't be part of this PR.

Comment (joe-wood-opentext) [reply helm/release/idol-omnigroupserver-0.6.0.tgz:1]: https://github.com/opentext-idol/idol-containers-toolkit/pull/63/commits/a8e7e95899a9f3fbe66737a7bca0deb34547047e

File: helm/idol-answerserver/templates/_container.tpl (hunk 2 above)
Comment (joe-wood-opentext) [reply helm/idol-answerserver/templates/_container.tpl]: https://github.com/opentext-idol/idol-containers-toolkit/pull/63/commits/c4d6d663accc17e2d6b5659d9f067f13c5d5574a For the model cache I've used the [default hugging face cache directory](https://teams.microsoft.com/l/message/19:e1d6a258680c4c73a25cdd5cfb9097bd@thread.tacv2/1721293683018?tenantId=10a18477-d533-4ecd-a78d-916dbd849d7c&groupId=bc988af4-1562-4554-bb99-ec7ac30b1738&parentMessageId=1721293683018&teamName=IDOL%20Text&channelName=F456017%20Search%20Abstractor%20conversations&createdTime=1721293683018), replacing `~` with `/home/idoluser` because `~` doesn't look like it works. Did you want this to be configurable via `values.yaml`?

File: helm/idol-answerserver/templates/_container.tpl [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+{{- $component := get . "component" | required "idolanswerserver.container: missing component" -}}
+{{- include "idol-library.util.merge" (dict
+  "root" $root
+  "component" $component
+  "source" "idol-library.aciserver.container.base.v1"
+  "destination" "idolanswerserver.container.base" 
+  "env" (list (dict "name" "IDOL_COMMAND_PARAMS" "value" "-idolcluster"))
+  "volumeMounts" (list (dict "name" "state" "mountPath" "/answerserver/license" "subPath" "license")
+                       (dict "name" "state" "mountPath" "/answerserver/uid" "subPath" "uid")
+                       (dict "name" "state" "mountPath" "/home/idoluser/.cache/huggingface" "subPath" "huggingface"))
```
Comment (daniel-lau-mf) [inline helm/idol-answerserver/templates/_container.tpl:30]:
```suggestion
                       (dict "name" "state" "mountPath" "/home/idoluser/.cache" "subPath" "cache"))
```
Doesn't need to be configurable, and probably more future flexible to keep the entire `.cache` dir

Comment (joe-wood-opentext) [reply helm/idol-answerserver/templates/_container.tpl:30]: Is this supposed to be `"subPath" ".cache"`?

```suggestion
                       (dict "name" "state" "mountPath" "/home/idoluser/.cache" "subPath" ".cache"))
```
Doesn't need to be configurable, and probably more future flexible to keep the entire `.cache` dir

**Review by daniel-lau-mf — APPROVED** [review]
